src/main/java/mServer/crawler/sender/arte/ArteVideoDetailsDeserializer.java

```diff
@@ -0,0 +1,194 @@
+package mServer.crawler.sender.arte;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import java.lang.reflect.Type;
+import java.text.ParseException;
+import java.util.Calendar;
+import mServer.tool.DateWithoutTimeComparer;
+import org.apache.commons.lang3.time.FastDateFormat;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class ArteVideoDetailsDeserializer implements JsonDeserializer<ArteVideoDetailsDTO> {
+
+    /*
+     * Grobe Struktur des Json's:
+     * {
+     *  "meta": {}
+     *  "programs": [{                   (JSON_ELEMENT_BROADCAST_ELTERNKNOTEN_1)
+          "broadcastProgrammings": [{    (JSON_ELEMENT_BROADCAST_ELTERNKNOTEN_2)
+            "broadcastBeginRounded": "2016-07-06T02:40:00Z",     (JSON_ELEMENT_BROADCAST)
+            "broadcastType": MAJOR_REBROADCAST, (JSON_ELEMENT_BROADCAST_TYPE)
+          ]}
+        ]}
+     * }
+     */    
+    private static final String JSON_ELEMENT_BROADCAST_ELTERNKNOTEN_1 = "programs";
+    private static final String JSON_ELEMENT_BROADCAST_ELTERNKNOTEN_2 = "broadcastProgrammings";
+    private static final String JSON_ELEMENT_BROADCAST = "broadcastBeginRounded";
+    private static final String JSON_ELEMENT_BROADCASTTYPE = "broadcastType";
+    private static final String JSON_ELEMENT_BROADCAST_CATCHUPRIGHTS_BEGIN = "catchupRightsBegin";
+    private static final String JSON_ELEMENT_BROADCAST_CATCHUPRIGHTS_END = "catchupRightsEnd";
+    private static final String BROADCASTTTYPE_FIRST = "FIRST_BROADCAST";
+    private static final String BROADCASTTTYPE_MINOR_RE = "MINOR_REBROADCAST";
+    private static final String BROADCASTTTYPE_MAJOR_RE = "MAJOR_REBROADCAST";
+    
+    private static final Logger LOG = LogManager.getLogger(ArteVideoDeserializer.class);
+
+    private final FastDateFormat broadcastDateFormat = FastDateFormat.getInstance("yyyy-MM-dd'T'HH:mm:ssX");//2016-10-29T16:15:00Z
+
+    private final Calendar today;
+
+    public ArteVideoDetailsDeserializer(Calendar aToday) {
+        today = aToday;
+    }
+    
+    @Override
+    public ArteVideoDetailsDTO deserialize(JsonElement aJsonElement, Type aType, JsonDeserializationContext aContext) throws JsonParseException {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 35 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)